pysteps/blending/steps.py

```diff
@@ -557,22 +572,19 @@ def forecast(
         fft,
     )
 
-    # 2.2 If necessary, decompose (NWP) model forecasts and stack cascades
-    (
-        precip_models_cascade,
-        mu_models,
-        sigma_models,
-        precip_models_pm,
-    ) = _compute_cascade_decomposition_nwp(
-        precip_models, bp_filter, decompositor, recompositor, fft, domain
-    )
+    # 2.2 If necessary, recompose (NWP) model forecasts
+    precip_models_cascade = None
+
+    if precip_models.ndim != 4:
+        precip_models_cascade = precip_models
+        precip_models = _compute_cascade_recomposition_nwp(precip_models, recompositor)
 
     # 2.3 Check for zero input fields in the radar and NWP data.
     zero_precip_radar = blending.utils.check_norain(precip, precip_thr, norain_thr)
     # The norain fraction threshold used for nwp is the default value of 0.0,
     # since nwp does not suffer from clutter.
     zero_model_fields = blending.utils.check_norain(
-        precip_models_pm, precip_thr, norain_thr
+        precip_models, precip_thr, norain_thr
```

Comment:
Why this name change? pm is only for the probability matching, or was the pm the only case of recombining the cascades, in which case this name is indeed better

Reply:
I don't know why it was called pm before, the only thing I know is that it (according to the comments in the code) would contain the recomposed model data. Recomposing the model data is the only way to get the input model data if you only supply the decomposed data, that is what I gathered from this code at least.

Reply:
Okay, I guess this is fine with the name change you made. I will also take this into account for the refactoring :) pm should, in general no longer be a thing I guess